homeassistant/components/hardware/websocket_api.py

```diff
@@ -2,23 +2,50 @@
 from __future__ import annotations
 
 import contextlib
-from dataclasses import asdict
+from dataclasses import asdict, dataclass
+from datetime import datetime, timedelta
+import importlib
+from typing import TYPE_CHECKING
 
 import voluptuous as vol
 
 from homeassistant.components import websocket_api
-from homeassistant.core import HomeAssistant, callback
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant, callback
 from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.event import async_track_time_interval
 
 from .const import DOMAIN
 from .hardware import async_process_hardware_platforms
 from .models import HardwareProtocol
 
+if TYPE_CHECKING:
+    import psutil  # noqa: F401
+
+
+@dataclass
+class SystemStatus:
+    """System status."""
+
+    psutil: psutil
+    remove_periodic_timer: CALLBACK_TYPE | None
+    subscribers: set[tuple[websocket_api.ActiveConnection, int]]
+
 
 @callback
 def async_setup(hass: HomeAssistant) -> None:
     """Set up the hardware websocket API."""
     websocket_api.async_register_command(hass, ws_info)
+    websocket_api.async_register_command(hass, ws_subscribe_system_status)
+    # Make a local copy of the psutil module because it relies on global variables
+    psutil_spec = importlib.util.find_spec("psutil")
+    assert psutil_spec and psutil_spec.loader
+    psutil_module = importlib.util.module_from_spec(psutil_spec)
+    psutil_spec.loader.exec_module(psutil_module)
```

Comment:
This is a workaround to cope with psutil storing state between calls in module global variables.

Alternatives could be to reimplement `psutil.cpu_percent` ourselves.

Reply:
Yeah it’s not so clean but the alternative would be to reimplement or monkey patch.

from a maintenance standpoint this has lower risk of breaking vs patching and it doesn’t add the additional burden of maintaining it ourselves

Reply:
It likely means system monitor is an issue as well though.  Maybe it makes sense to make a PyPI that does this under the hood so we get a clean copy for every integration that needs it so we only have this code in one place

Reply:
This is now exported to a library: https://github.com/home-assistant-libs/psutil-home-assistant